xgi/generators/nonuniform.py

```diff
@@ -288,3 +290,51 @@ def dcsbm_hypergraph(k1, k2, g1, g2, omega):
 
     df = pd.DataFrame(bipartite_edges)
     return Hypergraph(df)
+
+
+def random_hypergraph(N, ps):
+    """Generates a random hypergraph
+
+    Generate N nodes, and connect any d+1 nodes
+    by a hyperedge with probability ps[d-1].
+
+    Parameters
+    ----------
+    N : int
+        Number of nodes
+    ps : list of float
+        List of probabilities (between 0 and 1) to create a
+        hyperedge at each order d between any d+1 nodes. For example,
+        ps[0] is the wiring probability of any edge (2 nodes), ps[1]
+        of any triangles (3 nodes).
+
+    Returns
+    -------
+    Hypergraph object
+        The generated hypergraph
+
+    """
+
+    if (np.any(np.array(ps) < 0)) or (np.any(np.array(ps) > 1)):
+        raise ValueError("All elements of ps must be between 0 and 1 included.")
+
+    # I first generate a standard ER graph with edges connected with probability p1
+    G = nx.fast_gnp_random_graph(N, ps[0], seed=None)
```

Comment:
I agree that for the pairs that this is possibly faster, but I feel like we should use the same generator that you're using for the 3-hyperedges and larger for the whole hypergraph to a) be consistent andb) avoid importing NetworkX for this function only.

Reply:
I would also add a check for valid p values